subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -48,10 +48,11 @@ LOG_MODULE_REGISTER(bt_mesh_gatt);
 #define ADV_OPT_USE_NAME 0
 #endif
 
-#define ADV_OPT_PROXY                                                           \
-	(BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_SCANNABLE |                 \
-	 BT_LE_ADV_OPT_ONE_TIME | ADV_OPT_USE_IDENTITY |                       \
-	 ADV_OPT_USE_NAME)
+#define ADV_OPT_USE_NRPA(private) ((private) ? BT_LE_ADV_OPT_USE_NRPA : 0)
+
+#define ADV_OPT_PROXY(private)                                                                     \
+	(BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_SCANNABLE | ADV_OPT_USE_NRPA(private) |         \
+	 BT_LE_ADV_OPT_ONE_TIME | ADV_OPT_USE_IDENTITY | ADV_OPT_USE_NAME)
```

Comment:
Sorry to jump in a bit too late with this (the PR is already merged), but `USE_NRPA` and `USE_IDENTITY` sound conflicting to me. Perhaps `USE_IDENTITY` should be moved to the `ADV_OPT_USE_NRPA` macro expansion (instead of `0`) so that only one of those options is present in the final value?

Reply:
Should be fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/61592